src/lib.rs

```diff
@@ -872,3 +893,35 @@ macro_rules! exception {
         }
     };
 }
+
+/// Macro to set the function to be called at the beginning of the reset handler.
+///
+/// The function must have the signature of `unsafe fn()`.
+///
+/// The function passed will be called before static variables are initialized. Any access of static
+/// variables will result in undefined behavior.
+///
+/// # Examples
+///
+/// ``` ignore
+/// pre_init!(foo::bar);
+///
+/// mod foo {
+///     pub unsafe fn bar() {
+///         // do something here
+///     }
+/// }
+/// ```
+#[macro_export]
+macro_rules! pre_init {
+    ($handler:path) => {
```

Comment:
@yodaldevoid sorry to add yet another thing! But please could you change `path` to `expr` here, as per the entry/exception macros, so that it can accept a closure as well as a function path? The docstrings could then be updated accordingly too.

This changed in #78 for the other macros.

Reply:
So, I tried this when I was originally writing this as I saw that PR before it was merged, but this does not work as you cannot make an `unsafe` closure and the function passed needs to be unsafe as stated before by @japaric [here](https://github.com/rust-embedded/cortex-m-rt/pull/71#issuecomment-388630004).

Reply:
Ah, that's a shame. Fair enough.